src/box/replication.cc

```diff
@@ -316,6 +327,52 @@ replica_clear_id(struct replica *replica)
 	box_update_replication_synchro_quorum();
 }
 
+void
+replicaset_check_healthy_quorum(void)
```

Comment:
While experimenting with fencing I've stumbled upon following problem:
0) Having working replicaset with 3 replicas (quorum is 2 by default) I want to add another
1) I change box.cfg.replicaset on master node,  qourum becomes 3, new replica is not yet connected
2) replicase_check_healthy_quorum is called
3) replicaset_on_quorum_loss is called

IMO adding nodes to replicaset shouldn't lead to qourum_loss callback. I propose something like:
```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index 134546a51..a0982816e 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -1450,7 +1450,7 @@ box_update_replication_synchro_quorum(void)
        replication_synchro_quorum = quorum;
        txn_limbo_on_parameters_change(&txn_limbo);
        box_raft_update_election_quorum();
-       replicaset_check_healthy_quorum(1);
+       replicaset_check_healthy_quorum();
 }

 int
diff --git a/src/box/replication.cc b/src/box/replication.cc
index 559019542..6b604f2d8 100644
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -328,7 +328,7 @@ replica_clear_id(struct replica *replica)
 }

 void
-replicaset_check_healthy_quorum(int quorum_diff)
+replicaset_check_healthy_quorum(void)
 {
        /*
         * Quorum might be greater thn the number of registered nodes during
@@ -338,9 +338,9 @@ replicaset_check_healthy_quorum(int quorum_diff)
         */
        int quorum = MAX(replicaset.registered_count, 1);
        quorum = MIN(replicaset.registered_count, replication_synchro_quorum);
-       if (replicaset.healthy_count >= quorum && quorum_diff > 0)
+       if (replicaset.healthy_count >= quorum)
                trigger_run(&replicaset_on_quorum_gain, NULL);
-       else if (replicaset.healthy_count < quorum && quorum_diff < 0)
+       else
                trigger_run(&replicaset_on_quorum_loss, NULL);
 }

@@ -355,7 +355,7 @@ replica_add_applier_connection(struct replica *replica)
        replica->has_applier_connection = true;
        if (replica->has_relay_connection && !replica->anon) {
                replicaset.healthy_count++;
-               replicaset_check_healthy_quorum(1);
+               replicaset_check_healthy_quorum();
        }
 }

@@ -367,7 +367,7 @@ replica_remove_applier_connection(struct replica *replica)
        replica->has_applier_connection = false;
        if (replica->has_relay_connection && !replica->anon) {
                replicaset.healthy_count--;
-               replicaset_check_healthy_quorum(-1);
+               replicaset_check_healthy_quorum();
        }
        if (replica->id != 0)
                raft_notify_leader_seen(box_raft(), false, replica->id);
@@ -1014,7 +1014,7 @@ replica_add_relay_connection(struct replica *replica)
        replica->has_relay_connection = true;
        if (replica->has_applier_connection && !replica->anon) {
                replicaset.healthy_count++;
-               replicaset_check_healthy_quorum(1);
+               replicaset_check_healthy_quorum();
        }
 }

@@ -1026,7 +1026,7 @@ replica_remove_relay_connection(struct replica *replica)
        replica->has_relay_connection = false;
        if (replica->has_applier_connection && !replica->anon) {
                replicaset.healthy_count--;
-               replicaset_check_healthy_quorum(-1);
+               replicaset_check_healthy_quorum();
        }
 }

diff --git a/src/box/replication.h b/src/box/replication.h
index 1401a60e8..43b1cc186 100644
--- a/src/box/replication.h
+++ b/src/box/replication.h
@@ -438,7 +438,7 @@ replica_set_applier(struct replica * replica, struct applier * applier);
  *   during SUBSCRIBE.
  */
 void
-replicaset_check_healthy_quorum(int quorum_diff);
+replicaset_check_healthy_quorum(void);

 /** Track an established downstream connection in replica. */
 void
```